unstructured_ingest/v2/processes/connectors/pinecone.py

```diff
@@ -149,15 +155,28 @@ def precheck(self):
             raise DestinationConnectionError(f"failed to validate connection: {e}")
 
     @requires_dependencies(["pinecone"], extras="pinecone")
-    def upsert_batch(self, batch):
+    def upsert_batches_async(self, elements_dict: list[dict]):
         from pinecone.exceptions import PineconeApiException
 
-        try:
-            index = self.connection_config.get_index()
-            response = index.upsert(batch)
-        except PineconeApiException as api_error:
-            raise DestinationConnectionError(f"http error: {api_error}") from api_error
-        logger.debug(f"results: {response}")
+        chunks = list(
+            generator_batching_wbytes(
+                iterable=elements_dict,
+                batch_size_limit_bytes=MAX_PAYLOAD_SIZE - 100,
+                max_batch_size=self.upload_config.batch_size,
+            )
+        )
+        logger.info(f"Split doc with {len(elements_dict)} elements into {len(chunks)} batches")
+
+        pool_threads = self.upload_config.pool_threads or len(chunks)
```

Comment:
Couldn't `len(chunks)` be a big number? I would cap this at something like 10.

I'm really concerned about rate limiting here though and would like to default `pool_threads = 1`

We've had so many issues with pinecone failing and I would prefer we add error handling and retries before we optimize too much.

Reply:
Updated to use 1 thread by default.